gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -109,8 +109,7 @@ def self._check_final_ancestors(target, target_ancestors, source_method_names)
         # correctness, since ancestor.method_defined?(method_name) may return true even if method_name is not defined
         # directly on ancestor but instead an ancestor of ancestor.
         if (ancestor.method_defined?(method_name) ||
-            ancestor.private_method_defined?(method_name) ||
-            ancestor.protected_method_defined?(method_name)) &&
+            ancestor.private_method_defined?(method_name)) &&
            final_method?(method_owner_and_name_to_key(ancestor, method_name))
```

Comment:
OK, so doing this is (probably) a win, but maybe we don't have to do the `method_defined?` dance at all, we can just say:

`if final_method?(method_owner_and_name_to_key(ancestor, method_name))`

or at least check that *first*, because that's a much faster check than `method_defined?` -- I'm not really sure of the circumstances where the situation would be:

* `final_method?` returns true; but
* `method_defined?`/`private_method_defined?` returns false

I guess that if we *were* in such a situation, we wouldn't be diagnosing it currently, and that would be a problem.